scrapy/pqueues.py

```diff
@@ -53,43 +46,28 @@ def _scheduler_slot(request):
         url = request.get('url', None)
         slot = urlparse(url).hostname or ''
     elif isinstance(request, Request):
-        url = request.url
         slot = urlparse_cached(request).hostname or ''
 
+    # FIXME: meta is not modified when request is a dict and no meta is stored?
```

Comment:
To address this concern we should preservehttps://github.com/whalebot-helmsman/scrapy/blob/d8e2b25ba7811543d4be6c6bb6a8e851653bf368/scrapy/pqueues.py#L40 and replace `meta = request.get('meta', dict())` by `meta = request.setdefault('meta', dict())` inside this code